### DISCUSSION TITLE
[공지사항] pull request 할 때 조심할 점

### DISCUSSION THREAD

**Discussion by yoomambo** [4Moyede/HexatonClass-01#53] [00_공지사항]
여러분 안녕하세요

멘토 유혁준입니다.

@JaebeomHan 재범 멘티님께서 #51 이미 push한  commit에 파일을 추가하는 과정에서 생긴 문제 에 대한 이슈를 
스스로 해결하셨습니다! 문제를 혼자 해결하셔서 정말 다행이고, 대견하네요 ㅎㅎ

하지만 해결하시면서 약간의 문제점이 있었고, 이 분들은 멘티분들이 헷갈릴 수 있어 말씀드리려합니다.

Question : #47 #48 #49 #50 보시면 **_pull request를 직접 하셨습니다._** 이게 과연 어떤 문제가 될까요?

Hint : **pull request의 목적, 본질을 한 번 googling해서 찾아보시길 바랄게요.**

**Comment by ghost** [comment]
실제 작업에서 제가 짠 코드를 협업자들에게 보여주고 검토 후 병합할 수 있게끔 해야 하는데, 제가 직접 해버리면 그 과정들이 사라져 버리고 결국 검증되지 않은 코드들이 master branch에 들어오게 되는 문제점...? 제 이해가 맞을까요

**Reply by yoomambo** [reply]
안녕하세요 @JaebeomHan 멘티님 

멘토 유혁준입니다.

멘티님이 정확히 말씀해주셨네요.

보통 pull request를 보내면 협업자들과 토론 및 리뷰, 검증을 거친 후 main branch에 pull을 진행하게 됩니다.
그래서 꼭 pull request를 요청하시고 토론을 하는 단계가 필요합니다.

감사합니다.
